frontend/lib/with-signed-in.tsx

```diff
@@ -0,0 +1,24 @@
+import React from "react"
+import { NextPageContext as NextContext } from "next"
+import { isSignedIn } from "/lib/authentication"
+import redirect from "/lib/redirect"
+
+export default function withSignedIn(Component: any) {
+  return class WithSignedIn extends React.Component {
+    static displayName = `withSignedIn(${Component.displayName ?? "Component"})`
+
+    static async getInitialProps(context: NextContext) {
+      if (!isSignedIn(context)) {
+        redirect(context, "/sign-in")
+      }
+
+      return {
+        ...(await Component.getInitialProps?.(context)),
+      }
+    }
+
+    render() {
+      return <Component {...this.props}>{this.props.children}</Component>
```

Comment:
If the page transition happens to happen on client side, I believe this can render for a brief moment even if we're not signed in. To be safe I think we should render a simple placeholder here in that case e.g.
```jsx
if (!this.props.isSignedIn) {
  return <div>Redirecting...</div>
}
```

Reply:
Fixed